### DISCUSSION TITLE
Migrazione HA da venv a docker

### DISCUSSION THREAD

**Discussion by Gerrino1965** [indomus/forum#2495] [HUB: Home Assistant]
Ho la seguente installazione su Raspberry Pi OS:
Home Assistant Core in ambiente venv con:
- dehydrated
- HACS
- Integrazione Shelly
- Integrazione Sonoff
- Integrazione NUT
- Integrazione Node-RED Companion
Ho installato pure:
- InfluxDB
- Grafana
- Node-Red
Sono tentato di provare nuovamente la migrazione di Ha su docker per il fastidio del periodico aggiornamento di python, effetuato più volte seguendo la guida https://indomus.it/guide/installare-python-3-10-x-o-superiori-su-raspberry-pi-os-e-ripristinare-home-assistant-core-su-venv/, ma mai in maniera indolore.
E' possibile migrare in maniera indolore il solo HA?
Se si, quali precauzioni prendere oltre fare il backup e restore nella nuova directory di configurazione di .storage?

**Comment by snakuzzo** [comment]
La migrazione su Docker non è complicata.
Per quanto riguarda Core, ti basta creare il container configurando il bind della directory di configurazione attuale...e dopo correggere nei file di configurazione eventuali path o piccole cose qua e là...nulla che non ti appaia eventualmente nei log come WARNING o ERROR.

Quanto al resto, premesso che non è detto che tu debba "dockerizzare" tutto il resto, per pulizia puoi pensare di farlo:
- Shelly -> è un'integrazione presente già su Core, quindi non devi fare nulla
- Sonoff -> Non capisco a cosa tu ti riferisca, ma immagino sia un custom component...e quindi fa parte già della configurazione di Core che hai tu
- NUT -> immagino tu ti riferisca alla parte server, in tal caso guarda [QUI](https://indomus.it/guide/gestire-un-ups-con-raspberry-pi-tramite-network-ups-tools-nut-con-docker-su-raspbian/)
- Node-Red Companinont -> anche questa è un custom
- InfluxDB e Grafana -> se guardi su dockerhub, o sui siti ufficiali dei rispettivi progetti, ci sono tutte le guide per creare i container
- Node-Red -> Guarda [QUI](https://indomus.it/news/node-red-come-installarlo-e-configurarlo-su-linux-debian-con-docker/)
- Dehydrated -> Se cerchi su Dockerhub ci sono diverse immagini disponibili

**Reply by Gerrino1965** [reply]
Intanto grazie della risposta.
A parte l'errore mio, non sonoff ma tasmota, un'integrazione per la quale suppongo valga ciò che scrivi della shelly, quello che volevo sapere è se il resto è possibili lasciarlo così com'è. Fuori da un container docker. La mia necessità è solo evitare l'aggiornamento di python che non mi è mai riuscito in maniera indolore. Quello che funziona lo lascerei così com'è

**Reply by snakuzzo** [reply]
Sì certo, puoi farlo tranquillamente

**Reply by Gerrino1965** [reply]
Nessun accorgimento particolare nella configurazione di docker-compose? HA continuerà a cercare le porte dei vari servizi, che girano sullo stesso host fisico, fuori dal container? Per esempio la 8086 per influxdb?

**Reply by snakuzzo** [reply]
Guarda, devi studiare bene come funzionano le configurazioni di network su docker. Ad ogni modo, se segui le indicazioni sul sito ufficiale o nelle nostre guide, il container di HA viene creato con network mode host, quindi a livello di rete non c'è l'isolation di docker...utilizza quindi la stessa network dell'host
